server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1699,6 +1700,10 @@ public QueryRewriteContext getRewriteContext(LongSupplier nowInMillis) {
         return new QueryRewriteContext(parserConfig, client, nowInMillis);
     }
 
+    public DataRewriteContext getDataRewriteContext(LongSupplier nowInMillis) {
+        return new DataRewriteContext(parserConfig, client, nowInMillis);
+    }
```

Comment:
Wouldn't calling this from here still return a non-null value on a coordinator node?

Reply:
Possibly. Its a "trust" bases. I think the same goes for the coordinator rewrite, it could be constructed on the data node :/